tests/kafkatest/tests/client/consumer_test.py

```diff
@@ -123,9 +123,10 @@ def test_broker_rolling_bounce(self, metadata_quorum=quorum.zk, use_new_coordina
         #       nodes have time to expire
         self.rolling_bounce_brokers(consumer, clean_shutdown=True)
 
-        unexpected_rebalances = consumer.num_rebalances() - num_rebalances
-        assert unexpected_rebalances == 0, \
-            "Broker rolling bounce caused %d unexpected group rebalances" % unexpected_rebalances
+        if group_protocol == consumer_group.classic_group_protocol:
+            unexpected_rebalances = consumer.num_rebalances() - num_rebalances
+            assert unexpected_rebalances == 0, \
+                "Broker rolling bounce caused %d unexpected group rebalances" % unexpected_rebalances
```

Comment:
The [current framework](https://github.com/apache/kafka/blob/0a4a12fbc4df0ac4b141a869d3880b15b66a7759/tests/kafkatest/services/verifiable_consumer.py#L454-L456) calculates `num_rebalances` by increment by one every time a new assignment is received.

For new protocol consumer, if reconciliation happens during bouncing the brokers, `num_rebalances` will also be incremented. Currently there's no way to figure out whether a new rebalance is triggered or a reconciliation happens, so the assertion is skipped here.

Reply:
I think it makes sense to make sure the group is fully reconciled before taking num_rebalances. I'll add a method for it.